.github/workflows/ci-cd.yml

```diff
@@ -195,11 +200,19 @@ jobs:
         path: dist
 
     - name: Setup Python ${{ matrix.pyver }}
-      id: python-install
+      if: >-
+        !endsWith(matrix.pyver, '-freethreading')
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.pyver }}
         allow-prereleases: true
+    - name: Setup Python ${{ matrix.pyver }}
+      if: endsWith(matrix.pyver, '-freethreading')
+      uses: deadsnakes/action@v3.2.0
+      with:
+        python-version: 3.13-dev
+        nogil: true
```

Comment:
```suggestion
        nogil: true
    - name: Compute runtime Python version
      id: python-install
      run: |
        from os import environ
        from pathlib import Path
        from sys import version_info

        FILE_APPEND_MODE = 'a'

        python_version_str = ".".join(map(str, sys.version_info[:3]))

        with Path(environ['GITHUB_OUTPUT']).open(
                mode=FILE_APPEND_MODE,
        ) as outputs_file:
            print(f'python-version={python_version_str}', file=outputs_file)
      shell: python
```

Reply:
Although, we should probably extend that with a free-threading flag or something.